types/htmlbars-inline-precompile/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for htmlbars-inline-precompile 1.0
+// Project: ember-cli-htmlbars-inline-precompile
+// Definitions by: Chris Krycho <https://github.com/chriskrycho>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// This is a bit of a funky one: it's from a [Babel plugin], but is exported for
+// Ember applications as the module `"htmlbars-inline-precompile"`. It acts
+// like a tagged string from the perspective of consumers, but is actually an
+// AST transformation which generates a function as its output. That function in
+// turn [generates a string or array of strings][output] to use with the Ember
+// testing helper `this.render()`.
+//
+// [Babel plugin]: https://github.com/ember-cli/babel-plugin-htmlbars-inline-precompile#babel-plugin-htmlbars-inline-precompile-
+// [output]: https://github.com/emberjs/ember-test-helpers/blob/77f9a53da9d8c19a85b3122788caadbcc59274c2/lib/ember-test-helpers/-legacy-overrides.js#L17-L42
+
+export default function hbs(tagged: TemplateStringsArray): string | string[];
```

Comment:
Sorry, to be as clear as possible: this is *not* a commonJS module.

The module I'm providing a type for here explicitly depends on the underlying babel transform (linked in the writeup as well as the opening PR comment), which itself depends on babel—as does Ember CLI, which is the only thing that uses this. Accordingly, there's no worry about whether users will be able to depend on having Babel. Moreover, TypeScript and Babel are well-integrated and play nicely with each other in that toolchain, the optional TypeScript component of which I maintain.

The only question is whether the TypeScript integration will fall over when it encounters an unknown import. That's what this typing is meant to address, and it works exactly as expected in that context.